src/test/regress/input/alter_db_set_tablespace.source

```diff
@@ -746,15 +746,81 @@ DROP TABLESPACE adst_source_tablespace;
 DROP TABLESPACE adst_destination_tablespace;
 SELECT force_mirrors_to_catch_up();
 
+-- End of tests which require create_restartpoint_on_ckpt_record_replay=on
+--- start_ignore
+\! gpconfig -r create_restartpoint_on_ckpt_record_replay --skipvalidation;
+\! gpstop -u;
+--- end_ignore
+
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+-- If a mirror checkpointer doesn't create checkpoint between making changes in a database and moving this database to another tablespace, then attempt to create checkpoint the
+-- next time should not result in the fsync error and restarting of checkpointer. The checkpointer must process all requests.
+
+--- start_ignore
+-- Set fsync on since we need to test the fsync code logic.
+\! gpconfig -c fsync -v on --skipvalidation;
+\! gpstop -u;
+
+CREATE OR REPLACE FUNCTION mirror0() RETURNS gp_segment_configuration.dbid%type
+AS $$
+    SELECT dbid FROM gp_segment_configuration WHERE content = 0 AND role = 'm'
+$$ LANGUAGE SQL EXECUTE ON MASTER;
+--- end_ignore
+
+-- Checkpointer must not start processing requests until database is moved to another tablespace
+SELECT gp_inject_fault('ckpt_loop_begin', 'infinite_loop', mirror0());
+
+-- Wait until checkpointer is ready to start processing requests
+SELECT gp_wait_until_triggered_fault('ckpt_loop_begin', 1, mirror0());
+
+
+SELECT setup();
+-- Create the source and destination tablespaces
+CREATE TABLESPACE adst_source_tablespace LOCATION :'adst_source_tablespace_location';
+CREATE TABLESPACE adst_destination_tablespace LOCATION :'adst_destination_tablespace_location';
+
+-- Create a database in the source tablespace
+CREATE DATABASE alter_db TABLESPACE adst_source_tablespace;
+
+-- Make a change in the database, return to the previous database, set search_path again
+\c alter_db
+CREATE TABLE t(i int) DISTRIBUTED RANDOMLY;
+\c regression
+SET search_path TO adst,public;
+
+-- Prepare fault for waiting
+SELECT gp_inject_fault('ckpt_loop_end', 'skip', mirror0());
+
+ALTER DATABASE alter_db SET TABLESPACE adst_destination_tablespace;
+
+-- Ensure that the mirrors have applied the filesystem changes
+SELECT force_mirrors_to_catch_up();
+
+-- Checkpointer starts processing requests
+SELECT gp_inject_fault('ckpt_loop_begin', 'reset', mirror0());
+
+-- Wait until checkpointer finishes processing requests. If the fsync error happens, then server closes the connection and this fault is never reached
+SELECT gp_wait_until_triggered_fault('ckpt_loop_end', 1, mirror0());
+
+
+-- Cleanup
+DROP FUNCTION mirror0();
+DROP DATABASE alter_db;
+DROP TABLESPACE adst_source_tablespace;
+DROP TABLESPACE adst_destination_tablespace;
+
+--- start_ignore
+\! gpconfig -r fsync --skipvalidation;
```

Comment:
The default fsync GUC value is true. Which value do you expect reset?

Reply:
> The default fsync GUC value is true. Which value do you expect reset?

Fixed